File: lab/python/arrays_e_listas/frequencia_de_elementos.py
def frequencia1(lista):
    if not lista:
        return None

    ja_foi = []
    for i in lista:
        if i in ja_foi:
            continue
        cont = 0
        for j in range(len(lista)):
            if lista[j] == i:
                cont += 1
        print(f"{i} aparece {cont}x")
        ja_foi.append(i)


def frequencia2(lista):
    if not lista:
        return None

    contagem = {}
    for i in lista:
        contagem[i] = contagem.get(i, 0) + 1

    for k, v in contagem.items():
        print(f"{k} aparece {v}x")


lista = [2, 3, 2, 5, 3, 2]
frequencia1(lista)
print()
frequencia2(lista)

File: lab/javascript/arrays_e_listas/frequencia_de_elementos.js
function frequencia1(lista) {
  if (!lista || lista.length === 0) return null;

  const jaFoi = [];
  for (let i of lista) {
    if (jaFoi.includes(i)) continue;

    let cont = 0;
    for (let j = 0; j < lista.length; j++) {
      if (lista[j] === i) cont++;
    }

    console.log(`${i} aparece ${cont}x`);
    jaFoi.push(i);
  }
}

function frequencia2(lista) {
  if (!lista || lista.length === 0) return null;

  const contagem = {};
  for (let i of lista) {
    contagem[i] = (contagem[i] || 0) + 1;
  }

  for (let k in contagem) {
    console.log(`${k} aparece ${contagem[k]}x`);
  }
}

// -------------------- Teste --------------------
const lista = [2, 3, 2, 5, 3, 2];
frequencia1(lista);
console.log();
frequencia2(lista);
